packages/widget-chat/src/containers/message-composer/index.test.js

```diff
@@ -1,22 +1,115 @@
+/* eslint-disable max-nested-callbacks */
 import React from 'react';
 import renderer from 'react-test-renderer';
 
-import MessageComposer from '.';
+import {Provider} from 'react-redux';
+import {applyMiddleware, combineReducers, compose, createStore} from 'redux';
+import thunk from 'redux-thunk';
+
+import {user, conversation, message} from '../../reducers';
+import sparkReducer from '../../modules/redux-spark/reducers';
+
+
+import ConnectedMessageComposer, {MessageComposer} from '.';
+let component, store;
 
 describe(`MessageComposer component`, () => {
-  const component = renderer.create(
-    <MessageComposer
-      placeholder="Message Placeholder"
-      value="This is a message"
-    />
-  );
-
-  it(`renders properly`, () => {
-    expect(component).toMatchSnapshot();
+  beforeEach(() => {
+    store = createStore(
+      combineReducers({
+        user,
+        conversation,
+        message,
+        spark: sparkReducer
+      }),
+      compose(applyMiddleware(thunk))
+    );
+
+    component = renderer.create(
+      <Provider store={store}>
+        <ConnectedMessageComposer
+          placeholder="Message Placeholder"
+          value="This is a message"
+        />
+      </Provider>
+    );
+  });
+
+  describe(`snapshot tests`, () => {
+    it(`renders properly`, () => {
+      const tree = component.toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it(`handles value change properly`, () => {
+      const value = `abc123`;
+      const event = {
+        target: {
+          value
+        }
+      };
+      let tree = component.toJSON();
+      expect(tree).toMatchSnapshot();
+      const textarea = tree.children[0];
```

Comment:
nitpick: I don't think you need to declare a new reference here since `textarea` is only ever referenced once. Just reference `tree.children[0].props.onChange(event)` directly.